ibis/backends/__init__.py

```diff
@@ -863,6 +864,10 @@ def __init__(self, *args, **kwargs):
         self._con_args: tuple[Any] = args
         self._con_kwargs: dict[str, Any] = kwargs
         self._can_reconnect: bool = True
+        # mapping of memtable names to their finalizers
+        self._finalizers = {}
+        self._memtables = weakref.WeakSet()
+        self._current_memtables = weakref.WeakValueDictionary()
```

Comment:
I _think_ we can reduce this state to just a single weakset and dict, mind if I push up a commit?

Reply:
Go for it!

Reply:
@jcrist I am still open to you pushing up a commit if you've already got it locally, otherwise, no big deal.